csaf-validation/src/main/kotlin/io/github/csaf/sbom/validation/tests/Tests.kt

```diff
@@ -260,6 +269,213 @@ object Test617MultipleScoresWithSameVersionPerProduct : Test {
+            ValidationSuccessful
+        } else {
+            ValidationFailed(
+                listOf(
+                    "The latest version should be $latestVersion but is ${doc.document.tracking.version}"
+                )
+            )
+        }
+    }
+}
+
+/**
+ * Implementation of
+ * [Test 6.1.17](https://docs.oasis-open.org/csaf/csaf/v2.0/os/csaf-v2.0-os.html#6117-document-status-draft).
+ */
+object Test6117DocumentStatusDraft : Test {
+    override fun test(doc: Csaf): ValidationResult {
+        return if (
+            !doc.document.tracking.version.isZeroVersionOrPreRelease ||
+                (doc.document.tracking.status == Csaf.Status.draft)
+        ) {
+            ValidationSuccessful
+        } else {
+            ValidationFailed(
+                listOf(
+                    "The latest version is a pre-release or \"zero\" version (${doc.document.tracking.version}) but the document status is ${doc.document.tracking.status}"
+                )
+            )
+        }
+    }
+}
+
+object Test6118ReleasedRevisionHistory : Test {
+    override fun test(doc: Csaf): ValidationResult {
+        val finalStatues = listOf(Csaf.Status.final, Csaf.Status.interim)
+        // Only final or interim documents are applicable
+        if (doc.document.tracking.status !in finalStatues) {
+            return ValidationSuccessful
+        }
+
+        // Otherwise, we need to check for the revision history
+        val zeroVersions =
+            doc.document.tracking.revision_history.filter {
+                it.number == "0" || SemVer.parseOrNull(it.number)?.major == 0
+            }
+
+        return if (zeroVersions.isEmpty()) {
+            ValidationSuccessful
+        } else {
+            ValidationFailed(
+                listOf(
+                    "The document is ${doc.document.tracking.status} but it contains the following revisions: ${zeroVersions.map { it.number }.joinToString(", ")}"
+                )
+            )
+        }
+    }
+}
+
+object Test6119RevisionHistoryEntriesForPreReleaseVersions : Test {
+    override fun test(doc: Csaf): ValidationResult {
+        val preReleaseVersions =
+            doc.document.tracking.revision_history.filter {
+                SemVer.parseOrNull(it.number)?.preRelease != null
+            }
+
+        return if (preReleaseVersions.isEmpty()) {
+            ValidationSuccessful
+        } else {
+            ValidationFailed(
+                listOf(
+                    "The document contains the following pre-release revisions: ${preReleaseVersions.map { it.number }.joinToString(", ")}"
+                )
+            )
+        }
+    }
+}
+
+/**
+ * Implementation of
+ * [Test 6.1.20](https://docs.oasis-open.org/csaf/csaf/v2.0/os/csaf-v2.0-os.html#6120-non-draft-document-version).
+ */
+object Test6120NonDraftDocumentVersion : Test {
+    override fun test(doc: Csaf): ValidationResult {
+        val finalStatues = listOf(Csaf.Status.final, Csaf.Status.interim)
```

Comment:
Typo: finalStatuses or finalStatus.

final is confusing because it can also be `interim`. Maybe use non-draft to match the wording of the standard